cypress/integration/jsEmployeesTableDOM.spec.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+describe('Employees table', () => {
+  beforeEach('Open site', () => {
+    cy.visit('/');
+  });
+
+  it('should sort by name ASC', () => {
+    cy.get('th').contains('Name').click();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(1)')
+      .contains('Airi Satou').should('exist');
+  });
+
+  it('should sort by name DESC', () => {
+    cy.get('th').contains('Name').dblclick();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(1)')
+      .contains('Zorita Serrano').should('exist');
+  });
+
+  it('should sort by position ASC', () => {
+    cy.get('th').contains('Position').click();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(2)')
+      .contains('Accountant').should('exist');
+  });
+
+  it('should sort by position DESC', () => {
+    cy.get('th').contains('Position').dblclick();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(2)')
+      .contains('Technical Author').should('exist');
+  });
+
+  it('should sort by office ASC', () => {
+    cy.get('th').contains('Office').click();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(3)')
+      .contains('Edinburgh').should('exist');
+  });
+
+  it('should sort by office DESC', () => {
+    cy.get('th').contains('Office').dblclick();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(3)')
+      .contains('Tokyo').should('exist');
+  });
+
+  it('should sort by age ASC', () => {
+    cy.get('th').contains('Age').click();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(4)')
+      .contains(20).should('exist');
+  });
+
+  it('should sort by age DESC', () => {
+    cy.get('th').contains('Age').dblclick();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(4)')
+      .contains(66).should('exist');
+  });
+
+  it('should sort by salary ASC', () => {
+    cy.get('th').contains('Salary').click();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(5)')
+      .contains('$98,540').should('exist');
+  });
+
+  it('should sort by salary DESC', () => {
+    cy.get('th').contains('Salary').dblclick();
+
+    cy.get('tbody > :nth-child(1) > :nth-child(5)')
+      .contains('$452,500').should('exist');
+  });
+
+  it('row should have class active after click', () => {
+    cy.get('tbody > :nth-child(1) > :nth-child(1)').click();
+    cy.get('tr').should('have.class', 'active');
+  });
+
+  it('should be able to edit name by double click', () => {
+    cy.get('body').then(($body) => {
+      cy.get('tbody > tr:nth-child(1) > td:nth-child(1)').dblclick()
+        .then(() => {
+          if ($body.find('[class="cell-input"]').length > 0) {
+            cy.get('[class="cell-input"]').clear().type('Ihor Oliinyk{enter}');
+            cy.contains('Ihor Oliinyk');
+          } else {
+            cy.get(' tbody > tr:nth-child(1) > td:nth-child(1)');
+          }
+        });
```

Comment:
```suggestion
   const inputText = 'Banana';
  
   cy.get('tbody > tr:nth-child(1) > td:nth-child(1)').as('firstCell');
   cy.get('@firstCell').dblclick();
   cy.get('.cell-input').clear().type(`${inputText}{enter}`);
   cy.get('@firstCell').should('have.text', inputText);
```

Reply:
@vpolets can we make tasks 'edit cell by double click' as required, not optional?

Reply:
removed